snuba/utils/streams/backends/kafka.py

```diff
@@ -617,24 +619,55 @@ def closed(self) -> bool:
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
 
 
+@dataclass
+class KafkaBrokerConfig:
+    config: Mapping[str, Any]
+
+
+def get_broker_config(bootstrap_servers: Sequence[str]) -> Mapping[str, Any]:
+    return {
+        "bootstrap.servers": ",".join(bootstrap_servers),
+    }
+
+
 def build_kafka_consumer_configuration(
```

Comment:
Is there any reason not to encapsulate the logic that loads the settings here? 
It seems that logic is spread over multiple scripts. What about having these methods here:
```
# This looks for the storage annd then falls back to the default
def build_kafka_consumer_config(storage_name: str, group_id: str ... , override_params: Mapping[str, Any]) -> Mapping [str: Any]

# This does not take in the storage
def build_default_kafka_consumer_config(group_id: str ... , override_params: Mapping[str, Any]) -> Mapping [str: Any]

def build_producer_config(storage: str, override_params: Mapping[str, Any]) -> Mapping[str: Any]
```

These methods would load the config, apply the overrides (ensuring we do not try to override the server name for integrity of our config), remove the None values, and return the packed dictionary.

Also I would add a type alias for the returned field:
`KafkaBrokerConfig = Mapping[str, Any]`

It is mostly cosmetics in this file but it makes every file depending on this referencing a specific type that conveys something about its semantics instead of being a bare mapping.